cli: move flag definitions to a separate package (cli refactor sequence 6/n)

knz opened the pull request:
PR split from https://github.com/cockroachdb/cockroach/pull/82020 to ease review.

knz committed:
cliflagcfg: new package containing common flag def logic

This will be needed to ensure `cockroach-sql` supports the same env
vars.

Release note: None

knz committed:
cli/flag: move the flag definition logic to cliflagcfg

This ensures that the basic primitives are reusable out of package
`cli`.

Release note: None

knz committed:
cli: use cliflagcfg.FlagSetForCmd directly

Release note: None

knz committed:
cliflags: new package with the common client flag definitions

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/82141)
<!-- Reviewable:end -->

knz committed:
cliflagcfg: new variants of flag fns with depth arg

So that the caller can control at which level the env var
consumer check occurs.

(This can be explored via `cockroach debug env`.)

Release note: None
